shared_python/ValidateTags.py

```diff
@@ -0,0 +1,187 @@
+from logging import Logger
+
+class ValidateTags(object):
+    def __init__(self, log):
+        self.log = log
+
+    ## Dictionaries for tag validation
+    dict_tag_type = {
+        "rating": 1,
+        "warnings": 2,
+        "categories": 3,
+        "fandoms": 4,
+        "characters": 5,
+        "relationships": 6,
+        "tags": 7
+    }
+    dict_rating = {
+        "Not Rated": 1,
+        "General Audiences": 2,
+        "Teen And Up Audiences": 3,
+        "Mature": 4,
+        "Explicit": 5
+    }
+    dict_warning = {
+        "Choose Not To Use Archive Warnings": 1,
+        "Graphic Depictions Of Violence": 2,
+        "Major Character Death": 3,
+        "No Archive Warnings Apply": 4,
+        "Rape/Non-Con": 5,
+        "Underage": 6
+    }
+    dict_category = {
+        "Gen": 1,
+        "F/M": 2,
+        "M/M": 3,
+        "F/F": 4,
+        "Multi": 5,
+        "Other": 6
+    }
+
+    ## Functions for accessing dictionaries
+    def identify_tag_type(self, tag_type):
+        return self.dict_tag_type.get(tag_type, 0)
+
+    def identify_rating(self, tag):
+        return self.dict_rating.get(tag, 0)
+
+    def identify_warning(self, tag):
+        return self.dict_warning.get(tag, 0)
+
+    def identify_category(self, tag):
+        return self.dict_category.get(tag, 0)
+
+    ## Print statements
+    def print_tag_correction(self, before, after):
+        print('\r\033[96m', end='')
+        self.log.info('Correction successful. "' + before + '" is now "' + after+ '"')
```

Comment:
I'm definitely a fan of coloured output! I was going to say that Logger is a custom logger that already provides different colours, but it turns out I was thinking about [the fancy one in the eFiction code](https://github.com/otwcode/open-doors-eFiction/blob/master/opendoors/logging.py). However, we could do with having the same one in the ODAP anyway, and that would make the code here easier to read by just letting the custom Logger colour the messages according to their log level. But it can be a separate enhancement if you like as it's not specifically in scope for this feature.

Reply:
i'm interested in implementing that. it'll be nice to not have all these prints statements just to change and reset colors. But you're right, it's a little out of scope, and probably worth its own PR (I could make one for the sake of cleaning up code for the Tag Validator).

Reply:
Yeah, let's leave this as is and then you can follow up with a PR to add the fancy logger - it can even be its own Jira ticket to keep things separate.